main.cpp

```diff
@@ -354,19 +347,22 @@ Point::Point( const IntType& it) : Point(static_cast<float>(it), static_cast<flo
 
 Point& Point::multiply( const FloatType& m) 
 {
-    *this *= static_cast<float>(m);
+    x *= m;
+    y *= m;
     return *this;
 }
 
 Point& Point::multiply( const DoubleType& m)
 {
-    *this *= static_cast<float>(m);
+    x *= static_cast<float>(m);
+    y *= static_cast<float>(m);
     return *this;
 }
 
 Point& Point::multiply( const IntType& m) 
 {
-    *this *= static_cast<float>(m);
+    x *= static_cast<float>(m);
+    y *= static_cast<float>(m);
     return *this;
```

Comment:
- [ ] D.R.Y. all of this by restoring the original `Point::multiply(float)` function and call it in each of these.